src/kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs

```diff
@@ -241,18 +243,54 @@ impl<'tcx> GotocCtx<'tcx> {
     /// Currently, this is only proof harness annotations.
     /// i.e. `#[kani::proof]` (which kani_macros translates to `#[kanitool::proof]` for us to handle here)
     fn handle_kanitool_attributes(&mut self) {
-        let instance = self.current_fn().instance();
+        let attributes_list = self.tcx.get_attrs(self.current_fn().instance().def_id());
+        let (proof_attribute_vector, attribute_vector) = create_attribute_vectors(attributes_list);
+        let proof_attribute_count = proof_attribute_vector.len();
 
-        for attr in self.tcx.get_attrs(instance.def_id()) {
-            match kanitool_attr_name(attr).as_deref() {
-                Some("proof") => self.handle_kanitool_proof(),
-                _ => {}
+        // Handle the attributes only if there is one proof attribute
+        if proof_attribute_count == 1 {
+            self.create_proof_harness(attribute_vector);
+        }
+        // User error handling for user cases where there's more than one proof attribute being called
+        else if proof_attribute_count > 1 {
+            self.tcx
+                .sess
+                .span_warn(proof_attribute_vector[0].span, "Only one Proof Attribute allowed");
+        }
+        // User error handling for the user case where there's an attribute being called without #kani::tool
+        else if proof_attribute_count == 0 && attribute_vector.len() > 0 {
+            self.tcx.sess.span_err(
+                attribute_vector[0].1.span,
+                "Please use '#kani[proof]' above the annotation",
+            );
+        } else {
+            // No op for instances without any annotations declared
+        }
+    }
+
+    // Create the proof harness struct using the handler methods for various attributes
+    fn create_proof_harness(&mut self, attribute_vector: Vec<(String, &Attribute)>) {
+        // In the case when there's only one proof attribute (correct behavior), create harness and modify it
+        // depending on each subsequent attribute that's being called by the user.
+        let mut harness_metadata = self.handle_kanitool_proof();
+        if attribute_vector.len() > 0 {
+            // loop through all subsequent attributes
+            for attribute_tuple in attribute_vector.iter() {
+                // match with "unwind" attribute and provide the harness for modification
+                match attribute_tuple.0.as_str() {
+                    "unwind" => {
+                        self.handle_kanitool_unwind(attribute_tuple.1, &mut harness_metadata)
+                    }
+                    _ => {}
+                }
             }
         }
+        // self.proof_harnesses contains the final metadata that's to be parsed
+        self.proof_harnesses.push(harness_metadata);
     }
 
     /// Update `self` (the goto context) to add the current function as a listed proof harness
-    fn handle_kanitool_proof(&mut self) {
+    fn handle_kanitool_proof(&mut self) -> HarnessMetadata {
```

Comment:
Function doc comment needs updating. Also, we should rename this, maybe. "default_kanitool_proof" maybe?